src/GafferImage/CopyViews.cpp

```diff
@@ -0,0 +1,303 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferImage/CopyViews.h"
+
+#include "GafferImage/BufferAlgo.h"
+#include "GafferImage/ImageAlgo.h"
+
+#include "Gaffer/ArrayPlug.h"
+#include "Gaffer/Context.h"
+#include "Gaffer/NameValuePlug.h"
+
+#include "boost/bind/bind.hpp"
+#include "boost/range/adaptor/reversed.hpp"
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferImage;
+
+//////////////////////////////////////////////////////////////////////////
+// CopyViews
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_NODE_DEFINE_TYPE( CopyViews );
+
+size_t CopyViews::g_firstPlugIndex = 0;
+
+CopyViews::CopyViews( const std::string &name )
+	:	ImageProcessor( name, /* minInputs = */ 1 )
```

Comment:
Oh, I think I was just imitating one of the other Copy nodes - but I think it is reasonable to think that you could want to copy from multiple images, because this is how we implement merging views?

Reply:
Yeah, I think we do want an array here. I suppose according to the guidelines in the wiki, it should arguably be a `source` array plug, separate from the single `in` plug. The reasoning being that the logic for the array is different - everything is taken from `in`, but only the matching views are taken from `source`. I'm not sure how much value this has though - maybe for ImageProcessors the single array `in` is preferable? It's not like we have a separate array in `Merge`, just because the operation doesn't apply to the first image...